netty/src/main/java/io/grpc/transport/netty/NettyClientHandler.java

```diff
@@ -105,6 +128,59 @@ public void onGoAwayReceived(int lastStreamId, long errorCode, ByteBuf debugData
     });
   }
 
+  /**
+   * Sends a ping to the server.
+   */
+  void ping(final PingCallback callback, Executor executor) {
```

Comment:
Good call. I had a similar thought after I pushed this set of changes.

It should make the concurrency easier to reason about, too. Though one concern: if a callback is registered with `MoreExecutors.directExecutor()` it could potentially tie up one of Netty's NIO event loop threads. I guess that is (or should be) just a known risk of using direct executor though.